packages/widget.vite-plugin/src/build-plugin/buildWidgetSetManifest.ts

```diff
@@ -58,11 +60,36 @@ function buildWidgetManifest(
     entrypointCss: widgetBuild.stylesheets.map((path) => ({
       path: trimLeadingSlash(path),
     })),
-    parameters: widgetConfig.parameters,
+    parameters: convertParameters(widgetConfig.parameters),
     events: widgetConfig.events,
   };
 }
 
+function convertParameters(
+  parameters: Record<string, ParameterDefinition>,
+): Record<string, ManifestParameterDefinition> {
+  return Object.fromEntries(
+    Object.entries(parameters).map(([key, param]) => [
+      key,
+      convertParameter(param),
+    ]),
+  );
+}
+
+function convertParameter(
+  parameter: ParameterDefinition,
+): ManifestParameterDefinition {
+  if (parameter.type === "objectSet") {
+    return {
+      type: "objectSet",
+      displayName: parameter.displayName,
+      // Config has already been validated so rid must be present
+      objectTypeRids: [parameter.objectType.internalDoNotUseMetadata!.rid],
```

Comment:
Can we throw instead rather than silently allowing undefined through at runtime? The validation is defined elsewhere so would be good to quickly check that assumption here